test_pull_requests

```diff
@@ -935,6 +969,93 @@ popd
       end
     end
 
+    # has_valid_flake_comment? determines if a valid flake explanation comment exists,
+    # if there exists no valid comment, a human-readable reason why to help users
+    # understand the expected syntax
+    #
+    # A flake comment is valid iff:
+    #  - the comment contains the failed build URL
+    #  - all given issue links in the comment:
+    #    - reside in the correct repository
+    #    - point to issues with the correct label
+    def has_valid_flake_comment?(comments, build_url, settings)
+      general_issue_spec = /(https?:\/\/github.com\/.*\/issues\/[0-9]*)/
+      flake_issue_spec = /(https?:\/\/github.com\/#{Regexp.quote(Properties['github_user'])}\/#{Regexp.quote(settings['flake_identification']['repo'])}\/issues\/([0-9]*))/
+
+      valid_comment_exists = false
+      flake_label = settings['flake_identification']['label']
+      flake_repo = settings['flake_identification']['repo']
+      org_repo = %{#{Properties['github_user']}/#{flake_repo}}
+      flake_label_query = CGI.escape("label:#{flake_label}")
+      issue_link = %{https://github.com/#{org_repo}/issues?q=#{flake_label_query}}
+      reason = %{No comment was found that meets the flake identification requirements. A correctly formatted comment should include a link to the failed Jenkins job and at least one link to a GitHub issue in the [`#{org_repo}` repository](#{issue_link}) with the `#{flake_label}` label. An example of a comment with correct syntax would look like:
+```
+@#{Properties['bot_github_user']}, the [last build](#{build_url}) failed due to the following flakes:
```

Comment:
Still ugh.  On top of this impling something to users that isn't true.  Addressing the bot is sort of the wrong attitude.  The bot isn't going to fix the flakes for your.  Addressing the team would be more appropriate.